wa/workloads/uibenchjanktests/__init__.py

```diff
@@ -92,40 +108,69 @@ def __init__(self, target, **kwargs):
             instrument_wait=self.wait,
             no_hidden_api_checks=self.no_hidden_api_checks)
 
+    def validate(self):
+        if self.full and self.tests is not None:
+            raise ConfigError("Can't select subtests while 'full' is True")
+
     def initialize(self, context):
         super(Uibenchjanktests, self).initialize(context)
         self.dut_apk.initialize(context)
         self.dut_apk.initialize_package(context)
-        if 'class' not in self.apk.args:
-            class_for_method = dict(self.apk.apk_info.methods)
-            class_for_method[None] = self._DEFAULT_CLASS
-            try:
-                method = class_for_method[self.test]
-            except KeyError as e:
-                msg = 'Unknown test "{}". Known tests:\n\t{}'
-                known_tests = '\n\t'.join(
-                    m for m in class_for_method.keys()
-                    if m is not None and m.startswith('test'))
-                raise ValueError(msg.format(e, known_tests))
-            klass = '{}.{}'.format(self.package_names[0], method)
-
-            if self.test:
-                klass += '#{}'.format(self.test)
-            self.apk.args['class'] = klass
+
+        self.output = {}
+
+        # Full run specified, don't select subtests
+        if self.full:
+            self.apk.args['class'] = '{}.{}'.format(
+                self.package_names[0], self._DEFAULT_CLASS
+            )
+            return
+
+        self.available_tests = {
+            test: cl for test, cl in self.apk.apk_info.methods
+            if test.startswith('test')
+        }
+
+        # default to running all tests in alphabetical order
+        if not self.tests:
```

Comment:
Could you also add a `# pylint: disable=access-member-before-definition` for `self.tests` just to keep the linter happy [1]?

[1] https://github.com/ARM-software/workload-automation/actions/runs/5378640698/jobs/9758715470?pr=1216

Reply:
done, sure :)